File: 0757-set-intersection-size-at-least-two/0757-set-intersection-size-at-least-two.cpp
class Solution {
public:
    int intersectionSizeTwo(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b) {
            return (a[0] == b[0])? a[1] > b[1] : a[0] < b[0];});

        int ans = 0;
        vector<int> todo(intervals.size(), 2);

        while (!intervals.empty()) {
            auto x = intervals.back(); intervals.pop_back();
            int t = todo.back(); todo.pop_back();
            int s = x[0], e = x[1];

            for (int p = s; p < s + t; ++p) {
                for (int i = 0; i < intervals.size(); ++i) {
                    int s0 = intervals[i][0], e0 = intervals[i][1];
                    if (todo[i] > 0 && p <= e0)
                        --todo[i];
                }
                ++ans;
            }
            
        }

        return ans;
    }
};

File: 0757-set-intersection-size-at-least-two/0757-set-intersection-size-at-least-two.py
class Solution:
    def intersectionSizeTwo(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x: (x[0], -x[1]))
        todo = [2] * len(intervals)

        ans = 0

        while intervals:
            (s, e), t = intervals.pop(), todo.pop()

            for p in range(s, s + t):
                for i, (s0, e0) in enumerate(intervals):
                    if todo[i] and p <= e0:
                        todo[i] -= 1
                ans += 1

        return ans